atlasdb-config/src/main/java/com/palantir/atlasdb/config/AtlasDbConfig.java

```diff
@@ -313,17 +321,19 @@ private void checkNamespaceConfig() {
                 // (C* keyspace / Postgres dbName / Oracle sid). But changing the name of the TimeLock client
                 // will return the timestamp bound store to 0, so we also need to fast forward the new client bound
                 // to a value above of the original bound.
-                Preconditions.checkState(timeLockConfig.client().equals(keyValueServiceNamespace),
+                Preconditions.checkState(timeLockConfig.client().equals(Optional.of(keyValueServiceNamespace)),
                         "AtlasDB refused to start, in order to avoid potential data corruption."
                                 + " Please contact AtlasDB support to remediate this. Specific steps are required;"
                                 + " DO NOT ATTEMPT TO FIX THIS YOURSELF.");
             }
+            return keyValueServiceNamespace;
         } else if (timelock().isPresent()) {
             // Special case - empty timelock and empty namespace/keyspace does not make sense
-            boolean timelockClientNonEmpty = !timelock().get().client().orElse("").isEmpty();
-            Preconditions.checkState(timelockClientNonEmpty,
+            Preconditions.checkState(timelock().get().client().isPresent(),
                     "For InMemoryKVS, the TimeLock client should not be empty");
+            return timelock().get().client().get();
```

Comment:
nit: This last if block can probably be written `return timelock().get().client().orThrow(() -> new IllegalStateException("For InMemoryKVS..."));`. Up to you though

Reply:
Done.